### PR TITLE
[Form] Reverted string expectation in ChoiceToValueTransformer

### PR DESCRIPTION
**Pull request by walczakmac**
| Q | A |
| --- | --- |
| Branch? | 3.0 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18461 |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/ChoiceToValueTransformerTest.php [hunk 1]
```diff
@@ -78,10 +78,8 @@ public function testReverseTransform($in, $out, $inWithNull, $outWithNull)
     public function reverseTransformExpectsStringOrNullProvider()
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/ChoiceToValueTransformerTest.php]: You also need to rename those function with `*ExpectsScalarOrNull*`.

**Comment by HeahDude** [comment]
See also https://github.com/symfony/symfony/pull/17760#issuecomment-185739106 for a reason to not revert that change.

**Comment by HeahDude** [comment]
Another reason would be consistency with the `ChoicesToValuesTransformer` which implies the following:
```php
$builder->add('emailNotificationsEnabled', ChoiceType::class, [
    'required' => false,
    'multiple' => false,
    'choices'  => [0, 1],
    'empty_data' => 1, // works If we revert by merging this PR
]);
$builder->add('emailNotificationsEnabled', ChoiceType::class, [
    'required' => false,
    'multiple' => true, // ChoicesToValuesTransformer will be called
    'choices'  => [0, 1],
    'empty_data' => array(1), // won't work it expects array of strings array('1')
]);
```

**Comment by HeahDude** [comment]
And `empty_data => false` won't work either with the revert. Still need to pass `'0'`.

**Comment by HeahDude** [comment]
Also consider that:
```php
$builder->add('emailNotificationsEnabled', ChoiceType::class, [
    'required' => false,
    'multiple' => false,
    'choices'  => [null, 0, 1],
    'empty_data' => 1, // will match 0, needs the key '2' to match 1
]);

**Comment by HeahDude** [comment]
ref https://github.com/symfony/symfony-docs/issues/6340

**Comment by HeahDude** [comment]
Note that if accepted, this PR should be merged in 2.7.

**Comment by walczakmac** [comment]
Now I see that there are too many cons when reverting it to previous state, and it will be much easier if I use strings there instead of integers. I'll just close the PR and the ticket, also thanks for your help @HeahDude.

**Comment by HeahDude** [comment]
No worry ;) Thank you for the report.

**Pull request closed without merging**